### ISSUE TITLE
[Issue] - Health Checker - Check for dynamic memory has a rounding issue

### ISSUE THREAD

**Issue by thomass4t** [microsoft/CSS-Exchange#2203]
**Provide Version Number**
Version 24.09.17.1810

**Describe the issue**
When running the script, I receive the erorr message
"Dynamic Memory Detected: True 25.390625GB is the allowed dynamic memory of the server. Not supported to have dyn
amic memory configured."

Exchange is running as a Hyper-V machine and is not configured for dynamic memory.
The problem might arise from the PowerShell-Check:
`} elseif (($counter.CookedValue / 1024) -ne $totalPhysicalMemory) {`
$totalPhysicalMemory is 25 in our case and the left hand side results to 25,390625

**Expected behavior**
The error should not be shown if dynamic memory is disabled.

**Script Output**
see above snippet

**Additional context**
The Variable totalPhysicalMemory is already a rounded value:
`$totalPhysicalMemory = [Math]::Round($hardwareInformation.TotalMemory / 1MB)`
and shouldn't be compared with an unrounded float value.
$hardwareInformation.TotalMemory / 1024 / 1024 would be equal to $counter.CookedValue in my case (both exactly 26000)

**Comment by dpaulson45** [comment]
@thomass4t could you provide the xml and txt output from the Health Checker and a screen shot of the VM guest machine settings on the Memory tab to the email address of ExToolsFeedback@microsoft.com? That way I can look at to help determine what is going on and why yours is acting differently.

**Comment by thomass4t** [comment]
Logs are on the way via email ...

**Comment by thomass4t** [comment]
@dpaulson45 
Hello David, have you received the needed information? 
If yes, you might update the tags. Thank you and have a great week!

**Comment by dpaulson45** [comment]
@thomass4t yes, thank you for providing that data. I will try to review it and get back to you as soon as possible.

**Comment by worldsdream** [comment]
+1 it's also happening here. Showing dynamic memory detected on Hyper-V and I just double-checked that dynamic memory is DISABLED(unchecked). 

It shows in the log as "23.90625GB".
